components/net/http_loader.rs

```diff
@@ -1443,6 +1467,11 @@ fn http_network_fetch(
     // Step 16
 
     // Ensure we don't override "responseEnd" on successful return of this function
+    context
+        .timing
+        .lock()
+        .unwrap()
+        .set_attribute(ResourceAttribute::ResponseEnd);
```

Comment:
This change doesn't look right - there is no guarantee that the response is actually complete at this point. We already set this attribute during the future callbacks earlier this this method; why is this change necessary?

Reply:
My bad, I meant to write why in the commit message but I forgot. The reason was that one of the tests (/resource-timing/resource_TAO_origin.htm, I believe) was showing that ResponseEnd was still 0 even though it was supposed to be set to an actual number. I didn't have any luck getting the other calls to `set_attribute(ResourceAttribute::ResponseEnd)` to propagate the ResponseEnd value to the PerformanceResourceTiming interface so I thought adding it at the point shown above was the best approach.

Reply:
Aha, I see the issue - [this code](https://github.com/servo/servo/blob/06a07909c1dec52c9986db62a7a048efc0b42a05/components/net/http_loader.rs#L634) is where we associate the resource timing with a particular response object, but that doesn't take into account the fact that there could still be timing data changing. We should store an instance of `Arc<Mutex<ResourceFetchTiming>>` in Response instead of just `ResourceFetchTiming` to avoid this problem.

Reply:
We still need to remove this new set, since it happens too early.